### PR TITLE
Add `OPTIONS` for create index and constraint

### PR DESCRIPTION
**Pull request by Hunterness**
Builds on https://github.com/neo-technology/neo4j/pull/7435 which is merged

### CODE REVIEW

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala [hunk 1]
```diff
@@ -87,6 +87,15 @@ Create an index on the label `Person` and property `age` with the name `index_na
 ###assertion=create-index
 //
 
+CREATE INDEX FOR (p:Person) ON (p.surname)
+OPTIONS {indexProvider: 'native-btree-1.0', indexConfig: {`spatial.cartesian.min`: [-100.0, -100.0], `spatial.cartesian.max`: [100.0, 100.0]}}
+###
+
+Create an index on the label `Person` and property `surname` with the index provider `native-btree-1.0` and updated `spatial.cartesian` settings.
```
Comment (SaschaPeukert) [inline cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala]: Updated? So, it stays afterward?

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/ConstraintTest.scala [hunk 2]
```diff
@@ -163,6 +173,16 @@ or if the combination of the two is not unique,
...
 
+###assertion=create-node-key-constraint
+//
+
+CREATE CONSTRAINT node_key_with_config ON (p:Person)
+      ASSERT (p.name, p.age) IS NODE KEY
+      OPTIONS {indexConfig: {`spatial.wgs-84.min`: [-100.0, -100.0], `spatial.wgs-84.max`: [100.0, 100.0]}}
+###
+
+(★) Create a node key constraint on the label `Person` and properties `name` and `age` with the name `node_key_with_config` and updated `spatial.wgs-84` settings for the accompanying index.
```
Comment (SaschaPeukert) [inline cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/ConstraintTest.scala]: Updated? So, it stays afterward?

**Review by SaschaPeukert — COMMENTED** [review]
Please check my comments. If "updated" means "it is set as the new default afterward" you can merge this as it it. If not, please change it so it tells people that those configs are only temporary for the query they are used in

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala (hunk 1 above)
Comment (Hunterness) [reply cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala]: I think it is just for that index, but I don't really know how the config settings work.

@burqen Can you confirm that it is for this index only?

Comment (burqen) [reply cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala]: The index config is for the index (also for index backing a constraint) and I would say 'updated' is not the right word. It indicates to me that there already was some configuration for and existing index that we change, but the index is created with the given configuration.
Defining indexConfig for a single index creation does not change any default values.

I hope I interpreted your questions correctly ☺️

**Review by SaschaPeukert — APPROVED** [review]

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 3]
```diff
@@ -12,7 +12,11 @@ This section explains how to manage indexes used for search performance.
 * <<administration-indexes-examples, Examples>>
 ** <<administration-indexes-create-a-single-property-index, Create a single-property index>>
 ** <<administration-indexes-create-a-single-property-index-only-if-it-does-not-already-exist, Create a single-property index only if it does not already exist>>
+** <<administration-indexes-create-a-single-property-index-with-specified-index-provider, Create a single-property index with specified index provider>>
+** <<administration-indexes-create-a-single-property-index-with-specified-index-configuration, Create a single-property index with specified index configuration>>
+** <<administration-indexes-create-a-single-property-index-with-specified-index-configuration, Create a single-property index with specified index configuration>>
 ** <<administration-indexes-create-a-composite-index, Create a composite index>>
+** <<administration-indexes-create-a-composite-index-with-specified-index-provider-and-configuration, Create a composite index with specified index provider and configuration>>
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]: Why do i have 2 of the same?
```suggestion
** <<administration-indexes-create-a-single-property-index-with-specified-index-provider, Create a single-property index with specified index provider>>
** <<administration-indexes-create-a-single-property-index-with-specified-index-configuration, Create a single-property index with specified index configuration>>
** <<administration-indexes-create-a-composite-index, Create a composite index>>
** <<administration-indexes-create-a-composite-index-with-specified-index-provider-and-configuration, Create a composite index with specified index provider and configuration>>
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @Hunterness & @SaschaPeukert 👍 

This looks good.

**Pull request merged**